[DOCS] Adds 8.8.0 release notes (#155951)
elastic/kibana: docs/CHANGELOG.asciidoc

* Adds queues as nodes to the service map {kibana-pull}153784[#153784]
* Adds the ability to display the latest agent version in agent explorer {kibana-pull}153643[#153643]
* Adds table tabs showing summary of metrics {kibana-pull}153044[#153044]
* Adds warning to Edit Rule Flyout when publicUrl is not configured {kibana-pull}149832[#149832]

Dashboard::
Adds the ability to scroll the panel section {kibana-pull}145628[#145628]

Elastic Security::
For the Elastic Security 8.8.0 release information, refer to {security-guide}/release-notes.html[_Elastic Security Solution Release Notes_].

Enterprise Search::
For the Elastic Enterprise Search 8.8.0 release information, refer to {enterprise-search-ref}/changelog.html[_Elastic Enterprise Search Documentation Release notes_].

Fleet::
* Adds audit logging for core CRUD operations {kibana-pull}152118[#152118]
* Adds modal to display versions changelog {kibana-pull}152082[#152082]

Infrastructure::
* Adds the logs tab to the Hosts View {kibana-pull}152995[#152995]
* Adds Alerts tab into Hosts View {kibana-pull}149579[#149579]
* Adds refactoring to the Time and Position log stream state {kibana-pull}149052[#149052]

> Machine Learning::

Reviewer: ```suggestion
Machine Learning::
* Adds ELSER config to the Trained Models UI {kibana-pull}155867[#155867]
```

Machine Learning::
* DFA creation wizard: add ability to add time field to result data view {kibana-pull}155669[#155669]
* Display info when no datafeed preview results are found {kibana-pull}155650[#155650]
* Adding ignore unavailable indices option to anomaly detection job wizards {kibana-pull}155527[#155527]
* Support multiple model deployments {kibana-pull}155375[#155375]
* Uses two weeks before now for default start time in job start date picker {kibana-pull}155312[#155312]
* AIOps: Adds filter action for the Change point detection results {kibana-pull}155256[#155256]
* Adds search links for AIOps Labs pages {kibana-pull}155202[#155202]
* AIOps: Adds field stats for metric and split fields {kibana-pull}155177[#155177]
* AIOps: Link from Explain Log Rate Spikes to Log Pattern Analysis {kibana-pull}155121[#155121]
* Explain Log Rate Spikes: adds popover to analysis table for viewing other field values {kibana-pull}154689[#154689]
* Explain Log Rate Spikes: Makes use of random sampling for overall histogram chart {kibana-pull}154520[#154520]
* Explain Log Rate Spikes: Adds table action to copy filter to clipboard {kibana-pull}154311[#154311]

> * Data Frame Analytics: support custom urls in jobs {kibana-pull}154287[#154287]

Reviewer 2: This is already in the Features section. Is it OK to have it in here too?

Author: Fixed! Thank you for catching this.

* Change point detection: support for multiple metric and split fields {kibana-pull}154237[#154237]
* Enhances support for counter fields in data visualizer / field statistics {kibana-pull}153893[#153893]

> * Explain Log Rate Spikes: Support to filter fields from grouping {kibana-pull}153864[#153864]

Reviewer 2: This is already in the Features section. Is it OK to have it in here too?

Author: Fixed this one too.

[float]
[[fixes-v8.8.0]]
=== Bug Fixes
Alerting::
* Fixes Delete Schedule button padding issue {kibana-pull}154503[#154503]
* Fixes error message flash and throttle value reset {kibana-pull}154497[#154497]
* Fixes broken custom snooze recurrences with monthly frequency {kibana-pull}154251[#154251]
* Fixes an issue where you were unable to use retry on updateAPIKey conflict {kibana-pull}151802[#151802]

APM::
* Fixes an issue where you were uneable to enable framework alerts as data by default {kibana-pull}154076[#154076]
* Upgraded EUI to v76.0.0 {kibana-pull}152506[#152506]
* Fixes an issue where the OpenTelemetry process and system metrics were unsupported {kibana-pull}151826[#151826]

Canvas::
Fixes `createElement` callback {kibana-pull}154398[#154398]

Cases::
Fixes the Lens visualization in the comment and description markdown on the New Case page {kibana-pull}155897[#155897]

Dashboard::
* Fixes unsaved changes bug on empty dashboard {kibana-pull}155648[#155648]
* Removed Reload on Clone and Replace Panel {kibana-pull}155561[#155561]
* Fixes z index of toolbar items {kibana-pull}154501[#154501]
* Fixes inherited input race condition {kibana-pull}154293[#154293]

> * Fixes Changing label of a geospatial filter on causes filter disappear from map {kibana-pull}154087[#154087]

Reviewer 3: ```suggestion
* Fixes Changing label of a geospatial filter causes filter disappear from map {kibana-pull}154087[#154087]
```

* <<release-notes-8.1.0>>
* <<release-notes-8.0.0>>
* <<release-notes-8.0.0-rc2>>
* <<release-notes-8.0.0-rc1>>
* <<release-notes-8.0.0-beta1>>
* <<release-notes-8.0.0-alpha2>>
* <<release-notes-8.0.0-alpha1>>

--
[[release-notes-8.8.0]]
== {kib} 8.8.0

coming::[8.8.0]

Review the following information about the {kib} 8.8.0 release.

[float]
[[breaking-changes-8.8.0]]
=== Breaking changes

Breaking changes can prevent your application from optimal operation and performance.
Before you upgrade to 8.8.0, review the breaking changes, then mitigate the impact to your application.

// tag::notable-breaking-changes[]
[discrete]
[[breaking-155470]]
.Removes legacy project monitor API
[%collapsible]
====
*Details* +
The project monitor API for Synthetics in Elastic Observability has been removed. For more information, refer to {kibana-pull}155470[#155470].

*Impact* +
In 8.8.0 and later, an error appears when you use the project monitor API.
====
      
[discrete]
[[breaking-147985]]
.Changes the privileges for alerts and cases
[%collapsible]
====
*Details* +
The privileges for attaching alerts to cases has changed. For more information, refer to {kibana-pull}147985[#147985].

*Impact* +

> To attach alerts to cases, you must have `Read` access to Observability or Security, and `Write` access to the case. For detailed information, check link:https://www.elastic.co/guide/en/kibana/current/kibana-privileges.html[{kib privileges}] and link:https://www.elastic.co/guide/en/kibana/current/setup-cases.html[Configure access to cases]. 

Reviewer 4: ```suggestion
To attach alerts to cases, you must have `Read` access to Observability or Security, and `Write` access to the case. For detailed information, check link:https://www.elastic.co/guide/en/kibana/current/kibana-privileges.html[{kib} privileges] and link:https://www.elastic.co/guide/en/kibana/current/setup-cases.html[Configure access to cases]. 
```

> To attach alerts to cases, you must have `Read` access to Observability or Security, and `Write` access to the case. For detailed information, check link:https://www.elastic.co/guide/en/kibana/current/kibana-privileges.html[{kib} privileges] and link:https://www.elastic.co/guide/en/kibana/current/setup-cases.html[Configure access to cases]. 

Reviewer 4: If you prefer that I update the source PR, please just let me know since I'm not sure where this was derived from:
```suggestion
To attach alerts to cases, you must have `Read` access to an {observability} or Security feature that has alerts and `All` access to the **Cases** feature. For detailed information, check <<kibana-privileges.html>> and <<setup-cases>>. 
```

[%collapsible]
====
*Details* +
The PUT endpoint for the agent reassign API is deprecated. For more information, refer to {kibana-pull}152236[#152236].

*Impact* +
Use the POST endpoint for the agent reassign API.
====
      
[discrete]
[[deprecation-151564]]
.Deprecates `total` in `/agent_status` Fleet API
[%collapsible]
====
*Details* +
The `total` field in `/agent_status` Fleet API responses is deprecated. For more information, refer to {kibana-pull}151564[#151564].

*Impact* +
The `/agent_status` Fleet API now returns the following statuses:

* `all` &mdash; All active and inactive
* `active` &mdash; All active
====
      
[discrete]
[[deprecation-149506]]
.Deprecates Elastic Synthetics integration
[%collapsible]
====
*Details* +
The Elastic Synthetics integration is deprecated. For more information, refer to {kibana-pull}149506[#149506].

*Impact* +
To monitor endpoints, pages, and user journeys, go to **{observability}** -> **Synthetics (beta)**. 
====
      
[float]
[[features-8.8.0]]
=== Features
{kib} 8.8.0 adds the following new and notable features.

Alerting::
* Adds Maintenance Window Task Runner Integration + New AAD/Event Log Fields {kibana-pull}154761[#154761]

> * Adds support for file attachments in Cases {kibana-pull}154436[#154436]

Reviewer 5: I think all cases features should be in each own section. Cases can be used without alerts.

> * Adds support for file attachments in Cases {kibana-pull}154436[#154436]

Reviewer 6: Yep, I think we both agree on this @cnasikas. I can pull the kibana release notes down and make the change if you have no objections @KOTungseth @lcawl

> * Adds support for file attachments in Cases {kibana-pull}154436[#154436]

Reviewer 5: Hey, any update on this?

* Adds table tabs showing summary of metrics {kibana-pull}153044[#153044]
* Adds warning to Edit Rule Flyout when publicUrl is not configured {kibana-pull}149832[#149832]

Dashboard::
Adds the ability to scroll the panel section {kibana-pull}145628[#145628]

Elastic Security::
For the Elastic Security 8.8.0 release information, refer to {security-guide}/release-notes.html[_Elastic Security Solution Release Notes_].

Enterprise Search::
For the Elastic Enterprise Search 8.8.0 release information, refer to {enterprise-search-ref}/changelog.html[_Elastic Enterprise Search Documentation Release notes_].

Fleet::
* Adds audit logging for core CRUD operations {kibana-pull}152118[#152118]
* Adds modal to display versions changelog {kibana-pull}152082[#152082]

Infrastructure::
* Adds the logs tab to the Hosts View {kibana-pull}152995[#152995]
* Adds Alerts tab into Hosts View {kibana-pull}149579[#149579]
* Adds refactoring to the Time and Position log stream state {kibana-pull}149052[#149052]

Machine Learning::
* Adds support for custom URLs in jobs for Data Frame Analytics {kibana-pull}154287[#154287]

> * Adds support to filter fields from grouping in Explain Log Rate Spikes{kibana-pull}153864[#153864]

Reviewer 4: The lack of space is causing this to fail to resolve:
```suggestion
* Adds support to filter fields from grouping in Explain Log Rate Spikes {kibana-pull}153864[#153864]
```

> * Adds log pattern analysis in Discover {kibana-pull}153449[#153449]

Reviewer 7: The log pattern analysis would be a better fit in Discover, since it's only surfacing there, I think ... but of course it's also big news for Machine Learning, distributing their awesomeness in more parts of Kibana

Author: In the release blog, it lives under Machine Learning. But I don't have any strong opinions. Up to you and ML!

Reviewer 7: let's go with the release blog, we don't adorn ourselve with borrowed plumes 🪶

Reviewer 2: Can it be added to both Discover and Machine Learning? If not, then I agree with @kertal that it is better in Discover.

Reviewer 7: it's a big thing for both, so it would be worth a duplicate ... of we create a new section like **Disco Machine**  🕺🎰🤖💃

Author: New Disco Machine section. ON IT.

Author: Just kidding. I added it to both sections: ML and Discover.

Reviewer 7: Thx, **Disco🕺🎰  Lens 🤖💃 Machine** ... it will be in 9.0, FYI @ninoslavmiskovic because of the upcoming product branding !

* Adds Exploratory View to a separate app {kibana-pull}153852[#153852]

Platform::
Adds text {kibana-pull}151631[#151631]

Security::
* Adds CloudFormation agent install method {kibana-pull}155045[#155045]
* Adds Vul mgmt flyout details panel {kibana-pull}154873[#154873]
* Adds Vulnerabilities Table {kibana-pull}154388[#154388]
* Adds the ability to select a theme preference for {kib} in the User Profile {kibana-pull}151507[#151507]

Uptime::
Adds UUID to RuleAction {kibana-pull}148038[#148038]

For more information about the features introduced in 8.8.0, refer to <<whats-new,What's new in 8.8>>.

[[enhancements-and-bug-fixes-v8.8.0]]
=== Enhancements and bug fixes

For detailed information about the 8.8.0 release, review the enhancements and bug fixes.

[float]
[[enhancement-v8.8.0]]
=== Enhancements
Alerting::
* Adds the ability to predefine IDs when you create connectors {kibana-pull}155392[#155392]
* Adds the ability to allow the footer added to emails sent from {kib} alerting rules to **NOT** be added {kibana-pull}154919[#154919]
* Adds conditional actions UI for timeframe {kibana-pull}153944[#153944]

> * Adds the ability to filter user activities with pagination {kibana-pull}152702[#152702]

Reviewer 5: This should be in the `features` category. The PR had the wrong labels. I updated them accordingly.

Reviewer 5: We should add this PR in the enhancements section https://github.com/elastic/kibana/pull/155151. Sorry for the wrong labels.

* Enhances support for counter fields in data visualizer / field statistics {kibana-pull}153893[#153893]
* Explain Log Rate Spikes: Support to filter fields from grouping {kibana-pull}153864[#153864]
* Custom sorting by message level on Notifications page {kibana-pull}153462[#153462]
* Log pattern analysis in Discover {kibana-pull}153449[#153449]
* Explain Log Rate Spikes: Improves grouping using the `include` option of the `frequent_item_sets` agg {kibana-pull}153091[#153091]
* Data Frame Analytics exploration: adds actions column with link to discover {kibana-pull}151482[#151482]
* Allows row expansion for blocked anomaly detection jobs {kibana-pull}151351[#151351]
* Enhances job and datafeed config editors in the Advanced anomaly detection job wizard to provide suggestions and documentation {kibana-pull}146968[#146968]

Management::
* Adds timezone support for Transforms date histogram pivot configuration {kibana-pull}155535[#155535]
* Adds more system indices for storing internal data {kibana-pull}154888[#154888]
* Adds option to Reauthorize transform in Management page {kibana-pull}154736[#154736]

> * Adds secondary authorization header to Transforms in Fleet {kibana-pull}154665[#154665]

Reviewer 8: Maybe this should be grouped under `Machine Learning` or `Fleet`, I'm not sure. @qn895 WDYT?

Reviewer 9: @peteharverson I think `Adds option to Reauthorize transform in Management page` should be under Machine Learning, and `Adds user's secondary authorization to Transforms in Fleet` under Fleet. Wdyt?

Reviewer 2: `Adds option to Reauthorize transform in Management page` needs to go into the Management section (and not ML), as the Transforms feature is inside the Management page.

`Adds user's secondary authorization to Transforms in Fleet` should go under Fleet.

* Adds a _meta field to the Ingest pipelines form {kibana-pull}149976[#149976]

Maps::
Adds metrics mask {kibana-pull}154983[#154983]

Observability::
* Adds invalid license page {kibana-pull}154866[#154866]
* Adds empty state page links {kibana-pull}154678[#154678]
* Adds upload symbols instructions to add data page {kibana-pull}154670[#154670]
* Adds new CPU incl and CPU excl names {kibana-pull}154560[#154560]
* Adds symbols callout on frame information window {kibana-pull}154478[#154478]
* Adds Co2 and dollar cost columns and show more information action to functions table {kibana-pull}154097[#154097]
* Adds improvements to functions {kibana-pull}153873[#153873]
* Adds improvements to Flamegraph {kibana-pull}153598[#153598]
* Adds the ability to open the Traces view when you click on a series in stacked charts {kibana-pull}153325[#153325]
* Adds CPU usage column to replace CPU count column {kibana-pull}151696[#151696]

Querying & Filtering::
* Adds the ability to avoid duplicate host IP mapping {kibana-pull}155353[#155353]
* Adds improvements to the saved query terminology {kibana-pull}154517[#154517]

Security::

> * Adds the ability to configure remote index privileges directly from Role management {kibana-pull}154948[#154948]

Reviewer 10: ```suggestion
```

This feature was hidden just prior to feature freeze, so we shouldn't advertise it for this release.

> * Adds the Cloud_defend process index as a source and merged process event handling {kibana-pull}153213[#153213]

Reviewer 10: This probably belongs with the Elastic Security changelog

Author: @nastasha-solomon FYI

Reviewer 6: I'll make a note of this in the Security release notes doc issue to ensure it's included. Thanks for the ping, @KOTungseth!
cc: @benironside

[float]
[[fixes-v8.8.0]]
=== Bug Fixes
Alerting::
* Fixes Delete Schedule button padding issue {kibana-pull}154503[#154503]
* Fixes error message flash and throttle value reset {kibana-pull}154497[#154497]
* Fixes broken custom snooze recurrences with monthly frequency {kibana-pull}154251[#154251]
* Fixes an issue where you were unable to use retry on updateAPIKey conflict {kibana-pull}151802[#151802]

APM::
* Fixes an issue where you were uneable to enable framework alerts as data by default {kibana-pull}154076[#154076]
* Upgraded EUI to v76.0.0 {kibana-pull}152506[#152506]
* Fixes an issue where the OpenTelemetry process and system metrics were unsupported {kibana-pull}151826[#151826]

> 

Reviewer 5: Could you please add this PR https://github.com/elastic/kibana/pull/155897 in a `Cases` section?

* Fixes inherited input race condition {kibana-pull}154293[#154293]
* Fixes Changing label of a geospatial filter on causes filter disappear from map {kibana-pull}154087[#154087]

Discover::
* Adds a "Temporary" badge for temporary data views in the Alerts flyout {kibana-pull}155717[#155717]
* Adds the ability to exclude counter fields from Breakdown options {kibana-pull}155532[#155532]
* Adds the ability to skip requests for the time series metric counter field {kibana-pull}154319[#154319]
* Fixes KQL autocomplete suggestions, which now support IP-type fields when the `autocomplete:valueSuggestionMethod advanced setting is set to terms_enum {kibana-pull}154111[#154111]
* Fixes an issue where saved search "Manage searches" button was unable to apply the "search" type filter {kibana-pull}152565[#152565]

Elastic Security::
For the Elastic Security 8.8.0 release information, refer to {security-guide}/release-notes.html[_Elastic Security Solution Release Notes_].

Enterprise Search::
For the Elastic Enterprise Search 8.8.0 release information, refer to {enterprise-search-ref}/changelog.html[_Elastic Enterprise Search Documentation Release notes_].

Fleet::

> * Fixes package license check to use new constraints.elastic.subs… {kibana-pull}154831[#154831]

Reviewer 8: ```suggestion
* Fixes package license check to use new `conditions.elastic.subscription` field {kibana-pull}154831[#154831]
```

Monitoring::
Removes usage for the stats endpoint {kibana-pull}151082[#151082]

Observability::
* Adds space-specific feature privileges {kibana-pull}154734[#154734]
* Adds the ability to properly handle NO DATA with multiple conditions with a mix of aggregations and document count thresholds {kibana-pull}154690[#154690]
* Adds additional types to the fields to be use with cardinality aggregation for Metric Threshold Rule {kibana-pull}154197[#154197]
* Adds persistent normalization mode {kibana-pull}153116[#153116]
* Fixes refresh every in the alert search bar {kibana-pull}152246[#152246]

Platform::
Fixes badge counter for global settings {kibana-pull}150869[#150869]

Querying & Filtering::
* Adds the ability to unload a selected query when it is deleted {kibana-pull}154644[#154644]
* Removes failures in wrong custom timerange {kibana-pull}154643[#154643]

Reporting::
* Fixes report generation when image panel is in the end of the layout {kibana-pull}153846[#153846]
* Updates Chromium to 111.0.5555.0 (r1095492) and Puppeteer to 19.7.2 {kibana-pull}153033[#153033]

Security::

> Fixes Session view and k8s dashboard {kibana-pull}154982[#154982]

Reviewer 10: This probably belongs with the Elastic Security changelog as well

Author: @nastasha-solomon FYI

Reviewer 6: Thanks, @KOTungseth! I'll note this as well.

Platform::
New trigger actions for chart legends and table cell actions {kibana-pull}146779[#146779]

Querying & Filtering::
* Insight filter builder form as markdown plugin {kibana-pull}150363[#150363]
* Adds the ability to support complex filters with AND/OR relationships {kibana-pull}143928[#143928]

Security::
* Adds the ability to allow administrators to limit the number of concurrent user sessions with `xpack.security.session.сoncurrentSessions.maxSessions` {kibana-pull}147442[#147442]
* API Keys can now be updated with new role descriptors and metadata in the API Keys Management screen {kibana-pull}146237[#146237]

For more information about the features introduced in 8.7.0, refer to <<whats-new,What's new in 8.7>>.

[[enhancements-and-bug-fixes-v8.7.0]]
=== Enhancements and bug fixes
For detailed information about the 8.7.0 release, review the enhancements and bug fixes.

[float]
[[enhancement-v8.7.0]]
=== Enhancements
Alerting::
* Bring flapping status and settings in o11y {kibana-pull}150483[#150483]
* Redesign all cases list select modal {kibana-pull}149851[#149851]
* RenderCustomActionsRow with named params instead of args {kibana-pull}149304[#149304]

> * Adds new column `Updated on ` in `all cases list ` table. This column can be sorted and can persist sorting options {kibana-pull}149116[#149116]

Reviewer 5: Same as before, I believe all cases items should be in a separate category called `Cases`.
